jep/0000/README.adoc

```diff
@@ -0,0 +1,309 @@
+    - It allows classloading from agents to the controller (as before).
+    - It allows executing any callable regardless of its role check (as before).
+    - All `FileCallable` implementations, including those in `FilePath`, are allowed to act on any path.
+
+== Motivation
+
+The agent-to-controller security subsystem was added in 2014 to restrict the actions that agent processes can perform on the Jenkins controller as part of https://www.jenkins.io/security/advisory/2014-10-30/[the SECURITY-144 security fix].
+This protection was comprised of three major, complementary parts:
+
+Disabled class loading::
+Controllers do not load classes from agents, which means all code on a controller must already be part of that environment.
+No new code can be injected from agents.
+Role checks::
+Every `Callable` declares through its role check whether it's allowed to be sent from an agent to a controller.
+Legacy callables (built for Jenkins 1.565.3 or older, or Jenkins 1.586 or older) were rejected by default, but admins could allow their transmission from agents to the controller.
+File path filters::
+To continue supporting various methods on `FilePath` that transparently access files on the other side of a remoting channel, file path filters limit which files and directories can be accessed through them in what manner.
+
+=== Disabled class loading
+
+This is largely unchanged, except insofar as that there is no longer a UI option to disable it, just a Java system property escape hatch.
+
+=== Role checks
+
+`Callable` implementations can be separated into the following categories:
+
+* Implementations that allow their transmission from an agent to the controller (`SlaveToMasterCallable` or equivalent): These continue to be able to do this.
+* Implementations that prohibit their transmission from an agent to the controller (`MasterToSlaveCallable` or equivalent): Nothing changes, these always prohibited execution on the controller.
+* Implementations that do not perform a permission check (empty body of `#checkRoles(RoleChecker)`): https://www.jenkins.io/doc/upgrade-guide/2.303/#SECURITY-2458[A security hardening in Jenkins 2.319 and 2.303.3] prohibits this.
+* Implementations without a `RoleSensitive#checkRoles(RoleChecker)` implementation at all, in plugins built against Jenkins before 1.580.1 or 1.587:
+  These have always been prohibited unless on the allowlist (built-in or custom).
+
+With this proposal, the allowlist is removed, so any `Callable` that needed allowlisting to work will break.
+Few plugins should be affected, see below.
+
+In addition to `Callable`, `FileCallable` is an interface with equivalent role checks for use with `FilePath#act` (rather than `Channel#call`).
+The same four categories exist there.
+It should be noted that there is no need to allow their transmission in the agent-to-controller direction, this is just a convenient API for use with `FilePath#act`.
```

Comment:
Not sure I understood what you are saying here. #deprecation-of-slavetomasterfilecallable?

Reply:
Can probably just be removed in favor of that section and the explanation given there. I'm always struggling a bit with the JEP template.

Reply:
```suggestion
```